timemachine/fe/system.py

```diff
@@ -72,23 +82,27 @@ def simulate_system(U_fn, x0, num_samples=20000, steps_per_batch=500, num_worker
     return frames
 
 
-def convert_bps_into_system(bps):
+def convert_bps_into_system(bps: Sequence[potentials.BoundPotential]):
 
-    system = VacuumSystem(None, None, None, None, None, None)
+    bond = angle = torsion = nonbonded = None
 
-    for pot in bps:
-        if isinstance(pot, potentials.HarmonicBond):
-            system.bond = pot
-        elif isinstance(pot, potentials.HarmonicAngle):
-            system.angle = pot
-        elif isinstance(pot, potentials.PeriodicTorsion):
-            system.torsion = pot
-        elif isinstance(pot, potentials.Nonbonded):
-            system.nonbonded = pot
+    for bp in bps:
+        if isinstance(bp.potential, potentials.HarmonicBond):
+            bond = bp
+        elif isinstance(bp.potential, potentials.HarmonicAngle):
+            angle = bp
+        elif isinstance(bp.potential, potentials.PeriodicTorsion):
+            torsion = bp
+        elif isinstance(bp.potential, potentials.Nonbonded):
+            nonbonded = bp
         else:
             assert 0, "Unknown potential"
 
-    return system
+    assert bond
+    assert angle
+    assert nonbonded
```

Comment:
assert torsion?

Reply:
Torsion is optional in `VacuumSystem`, which is why it isn't asserted here (we have some code that constructs `VacuumSystem`s with `torsion=None`; I'm not sure if it's necessary, will double-check)